### PR TITLE
media-sound/snapcast: bump to 0.20.0, new proxy maintainer

### PR DESCRIPTION
**Pull request by jsmolic**
```diff
diff snapcast-0.15.0-r1.ebuild snapcast-0.20.0.ebuild
1c1
< # Copyright 1999-2018 Gentoo Authors
---
> # Copyright 1999-2020 Gentoo Authors
4c4,6
< EAPI="6"
---
> EAPI=7
>
> inherit cmake user
9,10c11,12
< if [[ ${PV} == *9999 ]] ; then
< 	inherit user cmake-utils git-r3
---
> if [[ ${PV} == 9999 ]] ; then
> 	inherit git-r3
15,16d16
< 	inherit user cmake-utils
<
18d17
<
24c23
< IUSE="+client +flac +server static-libs tremor +vorbis +zeroconf"
---
> IUSE="+client +expat +flac +opus +server static-libs tremor +vorbis +zeroconf"
28a28
> 	expat? ( dev-libs/expat )
29a30
> 	opus? ( media-libs/opus )
38,39d38
< PATCHES=( "${FILESDIR}/${PN}-options-for-use-flags.patch" )
<
53a53
> 		-DBUILD_WITH_EXPAT=$(usex expat)
54a55
> 		-DBUILD_WITH_OPUS=$(usex opus)
63c64
< 	cmake-utils_src_configure
---
> 	cmake_src_configure
67c68
< 	cmake-utils_src_install
---
> 	cmake_src_install
```

### CODE REVIEW

**Comment by thesamesam** [comment]
Need to rebase to make CI happy?

File: media-sound/snapcast/snapcast-9999.ebuild [hunk 1]
```diff
@@ -1,32 +1,33 @@
-# Copyright 1999-2018 Gentoo Authors
+# Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI="6"
```
Comment (thesamesam) [inline media-sound/snapcast/snapcast-9999.ebuild:4]: You're keeping the user eclass? Shock!

File: media-sound/snapcast/snapcast-9999.ebuild [hunk 2]
```diff
@@ -49,7 +50,9 @@ pkg_setup() {
 src_configure() {
 	local mycmakeargs=(
 		-DBUILD_CLIENT=$(usex client)
+		-DBUILD_WITH_EXPAT=$(usex expat)
 		-DBUILD_WITH_FLAC=$(usex flac)
+		-DBUILD_WITH_OPUS=$(usex opus)
 		-DBUILD_SERVER=$(usex server)
 		-DBUILD_STATIC_LIBS=$(usex static-libs)
 		-DBUILD_TESTS=no
```
Comment (thesamesam) [inline media-sound/snapcast/snapcast-9999.ebuild:58]: Do tests work?

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: media-sound/snapcast/snapcast-9999.ebuild (hunk 1 above)
Comment (jsmolic) [reply media-sound/snapcast/snapcast-9999.ebuild:4]: Ah, no :) I was waiting for someone to take a look at this first. I guess I'll have to make new acct packages to drop the eclass, that's what the CI is complaining about...

File: media-sound/snapcast/snapcast-9999.ebuild (hunk 2 above)
Comment (jsmolic) [reply media-sound/snapcast/snapcast-9999.ebuild:52]: I can't seem to find any tests, `make test` does nothing?? Weird, or I am missing something...

File: media-sound/snapcast/snapcast-9999.ebuild (hunk 1 above)
Comment (jsmolic) [reply media-sound/snapcast/snapcast-9999.ebuild:4]: Pushed new acct* packages

Comment (jsmolic) [reply media-sound/snapcast/snapcast-9999.ebuild:4]: side-note: I just saw now that new users and groups should be sent to gentoo-dev ML for review. I will do that in the following days.

**Comment by mjeveritt** [comment]
Please also file a PR against https://github.com/gentoo/api-gentoo-org/blob/master/files/uid-gid.txt

**Comment by juippis** [comment]
^ Please don't, we will give you a first free one during merging. It's less messy this way.

@ulm had to clean up the file because some numbers were reserved and never taken. Notet hat I haven't taken a look at this PR yet, but you can imagine a situation where it never gets to tree as-is, leaving the reserved numbers lying around.

**Comment by jsmolic** [comment]
@juippis Take your time with reviewing please
Long day at work when I was making this PR, somehow I saw that 461-463 numbers were free... numbers are changed now

**Comment by mjeveritt** [comment]
> ^ Please don't, we will give you a first free one during merging. It's less messy this way.
> 
Fair cop - glad you've got this procedure well-wrapped-up internally :+1: .

**Comment by ulm** [comment]
> @juippis Take your time with reviewing please
> Long day at work when I was making this PR, somehow I saw that 461-463 numbers were free... numbers are changed now

The main issue with these previous numbers was that they used different IDs for snapserver, namely UID 463 but GID 462. Policy explicitly says that this should be avoided:
https://projects.gentoo.org/qa/policy-guide/user-group.html#pg0901

(In principle, it is possible to choose a free UID where the same GID is used, or vice versa. That should be done only when it is certain that both packages will never add their corresponding UID/GID friends in the future.)

**Comment by jsmolic** [comment]
> > @juippis Take your time with reviewing please
> > Long day at work when I was making this PR, somehow I saw that 461-463 numbers were free... numbers are changed now
> 
> The main issue with these previous numbers was that they used different IDs for snapserver, namely UID 463 but GID 462. Policy explicitly says that this should be avoided:
> https://projects.gentoo.org/qa/policy-guide/user-group.html#pg0901
> 
> (In principle, it is possible to choose a free UID where the same GID is used, or vice versa. That should be done only when it is certain that both packages will never add their corresponding UID/GID friends in the future.)

Ah, I see, thanks for explaining. I wasn't aware of the policy at first so I chose the first free UID/GID for snapserver. Theoretically snapclient could then have UID 463... I changed it to 383 so we can leave it at that.

**Comment by jsmolic** [comment]
@juippis Can you please review? Thank you

**Comment by jsmolic** [comment]
ping

**Review by juippis — APPROVED** [review]

**Review by juippis — APPROVED** [review]

**Comment by jsmolic** [comment]
Thanks @juippis

**Comment by juippis** [comment]
Thanks for your patience, these GLEP-81 packages are a bit annoying to deal with :)

**Pull request closed without merging**